subsys/bluetooth/host/gatt.c

```diff
@@ -142,6 +142,9 @@ BUILD_ASSERT(!(CONFIG_BT_PERIPHERAL_PREF_TIMEOUT > 3200 &&
 BUILD_ASSERT((CONFIG_BT_PERIPHERAL_PREF_MIN_INT == 0xffff) ||
 	     (CONFIG_BT_PERIPHERAL_PREF_MIN_INT <=
 	     CONFIG_BT_PERIPHERAL_PREF_MAX_INT));
+BUILD_ASSERT((CONFIG_BT_PERIPHERAL_PREF_TIMEOUT * 4U) >
+	     ((1U + CONFIG_BT_PERIPHERAL_PREF_LATENCY) *
+	      CONFIG_BT_PERIPHERAL_PREF_MAX_INT));
```

Comment:
For those not familiar with the literal values here, where does the values `4U` and `1U` come from? 
Where does this requirement come from? 
If these are spec defined constraints, then I suggest adding a reference to the section in the specification, so this is not confused with a Zephyr-only constraint.

Reply:
It is also checked at the runtime in case someone tries to dynamically update connection parameters using Bluetooth API:

https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/hci_core.c#L1467

_/* All limits according to BT Core spec 5.0 [Vol 2, Part E, 7.8.12] */_

The requirement comes from the specification in the following form:

timeout > (latency + 1) * max_int * 2

However, timeout uses 10ms units and max_int uses 1.25ms units so the condition above needs to be converted to the following condition:

timeout' * 4 > (latency + 1) * max_int'

Reply:
Basically we have
```
timeout * 10 > (1 + latency * 1.25) * (interval * 1.25) * 2
```
before simplification, to convert to milliseconds. 

The simplification of just dividing all elements by 2.5 to get your equation is incorrect, you don't take the `1U` into account. 

What is the reason why you want to simplify it, instead of letting the compiler optimize the originally equation?

Reply:
@Thalley, latency does not have time units and its value should be the same in the spec condition and implemented condition (I indicated changed parameters with the ' mark in my generalized conditions).

Reply:
You are right, the latency is not in milliseconds. I was confused by the wording from Zephyr:
```
config BT_PERIPHERAL_PREF_LATENCY
	int "Peripheral preferred peripheral latency in Connection Intervals"
	default 0
	range 0 499
```
As `Connection Intervals` are `1.25ms`. What it should say is `number of connection events`. 

So with that, the equation looks correct. 

Sorry about the misunderstanding (should've use the core spec description and not Zephyr).